### PR TITLE
Jbaker/fix gradient

### PR DESCRIPTION
**Pull request by jason-capsule42**
# Alaska Airlines Pull Request

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Fix the gradient positioning in the carousel component and enhance it with theme support for customizable gradient colors. Update documentation to reflect these changes.

Bug Fixes:
- Fix the gradient positioning in the carousel component to ensure it appears behind the buttons without altering DOM order.

Enhancements:
- Add theme support for the carousel component, allowing customization of gradient colors through design tokens.

Documentation:
- Update documentation to include theme support information for the carousel component, detailing how to restyle using design tokens.

### CODE REVIEW

File: demo/api.md [hunk 1]
```diff
@@ -498,4 +498,21 @@ Used for all elements inside the scrolling carousel.
...
+
+### Theme Support
+
+The component may be restyled using the following code sample and changing the values of the following token(s).
+
+<!-- AURO-GENERATED-CONTENT:START (CODE:src=./../../src/tokens.scss) -->
+<!-- The below code snippet is automatically added from ./../../src/tokens.scss -->
+
+```scss
+@import './../node_modules/@aurodesignsystem/design-tokens/dist/tokens/SCSSVariables';
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply demo/api.md:511]: This doesn't work with markdown-magic

File: demo/api.min.js [hunk 2]
```diff
@@ -378,19 +1260,21 @@ class AuroCarousel extends LitElement {
...
+        aria-label="${to(this.label)}"
         aria-roledescription="carousel"
-        class="${classMap(carouselClassMap)}"
+        class="${Rt(carouselClassMap)}"
         @scroll=${() => this.setScrollFlags(true)} >
-        <${this.buttonTag} arialabel="chevron-left" iconOnly rounded variant="secondary" @click=${() => this.handleClick(false)} class="button button--left">
-          ${this.generateIconHtml(chevronLeft.svg)}<span class="util_displayHiddenVisually">Scroll Left</span>
-        </${this.buttonTag}>
         <div class="container">
+          <div class="gradient--left"></div>
```
[the remark this answers is not in the document]
Comment (jason-capsule42) [reply demo/api.min.js:1274]: Looking at changing all this to a `mark`. Not making this change at this time.

File: src/style.scss [hunk 3]
```diff
@@ -102,20 +102,38 @@
 
 .button--left {
   left: 0;
+  z-index: +1; // Ensure the gradient is behind the button without changing the order of the DOM elements
```
Comment (rmenner) [inline src/style.scss:105]: Never seen this before. I was trying to look it up. The `+` does not seem to serve any purpose.

**Review by rmenner — APPROVED** [review]

File: src/style.scss (hunk 3 above)
Comment (jason-capsule42) [reply src/style.scss:104]: +1 just means to treat it as high level higher in z-index than what it is by default. In this case, it bring's it above the gradient that is after it in the DOM.

Pull this down, delete the z-index rule, run the demo, scroll a carousel to the right (click on the right button) and you will see the issue on the left button being behind the gradient.

**Pull request merged**